### ISSUE TITLE
[Bug]: Copy/move to writable subfolder of write protected folder is not possible

### ISSUE THREAD

**Issue by bwurst** [nextcloud/server#44649]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When I write protect a folder (via low level file system permissions or with the ACLs of group folder app) and have a writeable folder inside it, this folder is not selectable on copy/move dialog on the web frontend.

I can copy files to that writeable subfolder if it is directly shared to me or if I use third party webdav access but not with the NC web interface. The issue is the same for regular folders or folders shared by someone else.

We need a fixed folder structure shared to our users and allow users to only write to deeper folders. Currently, they can upload and store files there but not move anything from their storage to the shared folder from NC UI.

### Steps to reproduce

1. create directory "folder" and "subfolder" inside it
2. go to low level files directory and issue "setfacl -m u:apache:rx folder ; setfacl -m u:apache:rwx folder/subfolder" (if PHP runs as apache)
3. rescan files with "occ files:scan"
4. move a arbitrary file or folder to "folder/subfolder", you cannot step into "folder" at all.

### Expected behavior

Folder should be visible, perhaps decorated with a write protection indicator.

### Installation method

Community Web installer on a VPS or web space

### Nextcloud Server version

28

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by joshtrichards** [comment]
You may have a legitimate point in the Web UI (if there is indeed a discrepancy versus when accessed via one of the clients - or even standard WebDAV), however I'm not clear why you think this would work:

> go to low level files directory and issue "setfacl -m u:apache:x folder ; setfacl -m u:apache:rwx folder/subfolder" (if PHP runs as apache)

Nextcloud expects full access throughout it's datadirectory. OS level adjustments like this aren't reflected in Nextcloud. At best, they do nothing. At worst, they interfere with Nextcloud's ability to enforce it's own permissions. Based on the example OS level ACL I would not expect Nextcloud to even be able to access `folder` after running the scan against it.

**Comment by bwurst** [comment]
Sorry, for the confusion, I made a mistake. The first ACL should include read-permission. I fixed it in the report.

The customer's wish is that arbitrary users cannot change the basic directory structure of a share but can modify files in subfolders.
This is not possible with plain nextcloud.
This can be done with advanced permissions in group folders app or file system level ACLs. Remote access works the same. The behaviour of nextcloud web UI is also the same. So I left out the group folders app to get a more minimalistic example.

Nextcloud internally can handle the situation just fine as permissions are saved for every folder. But the copy/move dialog of WebUI behaves bad with this.
